pkg/operator/controller/crl/crl_configmap.go

```diff
@@ -209,7 +220,7 @@ func desiredCRLConfigMap(ic *operatorv1.IngressController, ownerRef metav1.Owner
 	}
 	crlConfigmap.SetOwnerReferences([]metav1.OwnerReference{ownerRef})
 
-	return true, &crlConfigmap, nil
+	return true, &crlConfigmap, context.WithValue(ctx, "nextCRLUpdate", nextCRLUpdate), nil
```

Comment:
I went back and forth for a while on how to return `nextCRLUpdate` because it's kind of tangental to the main purpose of `desiredCRLConfigMap`, and adding that kind of extra return value for callers to manage feels like a bad design choice.

Ideally, `nextCRLUpdate` would be calculated in its own function to keep the code more clear, but doing that would require all the CRLs to be parsed a second time, and I understand that in certain scenarios, users are running up against the max size of a configmap (1MiB) for the CRL configmap, so parsing again could be pretty wasteful.

I think this is a reasonable compromise, where `nextCRLUpdate` is available for `Reconcile` to deal with it, but intermediate functions don't need to worry about it other than to pass the context back up.

Reply:
Sure, I can buy that, sounds like you just want to decouple `nextCRLUpdate` from the output of `desiredCRLConfigMap`.